cmds/cbfs/cbfs.go

```diff
@@ -1,37 +1,218 @@
 package main
 
 import (
+	"bytes"
+	"encoding/binary"
 	"fmt"
 	"log"
+	"os"
 
 	"github.com/linuxboot/cbfs/pkg/cbfs"
 	flag "github.com/spf13/pflag"
 )
 
-var debug = flag.BoolP("debug", "d", false, "enable debug prints")
+//CommandlineArgs holds the parsed command line arguments
+type CommandlineArgs struct {
+	Debug           bool
+	Image           string
+	Command         string
+	CustomArguments map[string]*CustomCommandlineArg
+}
 
-func main() {
-	flag.Parse()
+type CustomCommandlineArg struct {
+	Name  string
+	Short string
+	Usage string
+	Value string
+}
+
+var cmdArgFilename = CustomCommandlineArg{
+	Name:  "filename",
+	Short: "f",
+	Usage: "The file to operate on",
+}
+
+var cmdArgCBFSName = CustomCommandlineArg{
+	Name:  "name",
+	Short: "n",
+	Usage: "The name of the file in CBFS",
+}
+
+//CBFSCommand represents a command with arbitrary command lines options
+type CBFSCommand struct {
+	Name            string
+	Operation       func(args CommandlineArgs) error
+	CustomArguments []*CustomCommandlineArg
+}
+
+var cmds = []CBFSCommand{
+	{
+		Name: "list",
+		Operation: func(args CommandlineArgs) error {
+			i, err := cbfs.Open(args.Image)
+			if err != nil {
+				return fmt.Errorf("failed to open file %s. Error: %v", args.Image, err)
+			}
+
+			fmt.Printf("%s", i.String())
+			return nil
+		},
+	},
+	{
+		Name: "add-payload",
+		Operation: func(args CommandlineArgs) error {
+			if _, ok := args.CustomArguments["filename"]; !ok {
+				return fmt.Errorf("filename not set")
+			}
+			if _, ok := args.CustomArguments["name"]; !ok {
+				return fmt.Errorf("CBFS file name not set")
+			}
+
+			hdrs, data, err := cbfs.ELFToPayload(args.CustomArguments["filename"].Value, cbfs.LZ4)
+			if err != nil {
+				return fmt.Errorf("Failed to convert ELF to payload: %v", err)
+			}
+
+			buf := new(bytes.Buffer)
+			err = binary.Write(buf, binary.BigEndian, hdrs)
```

Comment:
Always check all errors.

Reply:
removed from this PR, feature will be added in the next PR as well as tests.